Client/Frontend/Settings/LoginDetailViewController.swift

```diff
@@ -99,47 +121,95 @@ class LoginDetailViewController: UITableViewController {
-                loginCell.highlightedLabel.text = NSLocalizedString("website", tableName: "LoginManager", comment: "Title for website row in Login Detail View")
-                loginCell.descriptionLabel.text = login.hostname
-                loginCell.enabledActions = [.Copy, .OpenAndFill]
-            }
+        let notificationCenter = NSNotificationCenter.defaultCenter()
+        notificationCenter.removeObserver(self, name: NotificationProfileDidFinishSyncing, object: nil)
+        notificationCenter.removeObserver(self, name: UIMenuControllerWillShowMenuNotification, object: nil)
+        notificationCenter.removeObserver(self, name: UIMenuControllerWillHideMenuNotification, object: nil)
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+
+        // The following hacks are to prevent the default cell seperators from displaying. We want to
+        // hide the default seperator for the website/last modified cells since the last modified cell
+        // draws it's own seperators. The last item in the list draws its seperator full width. The reason
+
+        // Prevent seperators from showing by pushing them off screen by the width of the cell
+        let itemsToHideSeperators: [InfoItem] = [.WebsiteItem, .LastModifiedSeperator]
+        itemsToHideSeperators.forEach { item in
+            let cell = tableView.cellForRowAtIndexPath(NSIndexPath(forRow: item.rawValue, inSection: 0))
+            cell?.separatorInset = UIEdgeInsetsMake(0, cell?.bounds.width ?? 0, 0, 0)
+        }
+
+        // Rows to display full width seperator
+        let itemsToShowFullWidthSeperator: [InfoItem] = [.DeleteItem]
+        itemsToShowFullWidthSeperator.forEach { item in
+            let cell = tableView.cellForRowAtIndexPath(NSIndexPath(forRow: item.rawValue, inSection: 0))
+            cell?.separatorInset = UIEdgeInsetsZero
+            cell?.layoutMargins = UIEdgeInsetsZero
+            cell?.preservesSuperviewLayoutMargins = false
+        }
+    }
+}
+
+// MARK: - UITableViewDataSource
+extension LoginDetailViewController: UITableViewDataSource {
+
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        switch InfoItem(rawValue: indexPath.row)! {
+        case .TitleItem:
+            let loginCell = dequeueLoginCellForIndexPath(indexPath)
+            loginCell.style = .IconAndDescriptionLabel
+            loginCell.descriptionLabel.text = login.hostname
             return loginCell
-        case .Delete:
-            let deleteCell = tableView.dequeueReusableCellWithIdentifier(DeleteCellIdentifier, forIndexPath: indexPath)
+
+        case .UsernameItem:
+            let loginCell = dequeueLoginCellForIndexPath(indexPath)
+            loginCell.style = .NoIconAndBothLabels
+            loginCell.highlightedLabel.text = NSLocalizedString("username", tableName: "LoginManager", comment: "Title for username row in Login Detail View")
```

Comment:
Guess we're not string frozen for now? I haven't really been able to keep up with our l10n status.